src/branch/main.rs

```diff
@@ -0,0 +1,30 @@
+use edgedb_tokio::get_project_dir;
+use crate::branch::context::Context;
+use crate::branch::{create, drop, list, rename, switch, wipe};
+use crate::branch::option::{BranchCommand, Command};
+use crate::connect::Connection;
+use crate::options::Options;
+use crate::portable::config::Config;
+
+#[tokio::main]
+pub async fn branch_main(options: &Options, cmd: &BranchCommand) -> anyhow::Result<()> {
+    let context = create_context().await?;
+
+    let mut connection: Connection = options.create_connector().await?.connect().await?;
+
+    // match commands that don't require a connection to run, then match the ones that do with a connection.
+    match &cmd.subcommand {
+        Command::Switch(switch) => switch::main(switch, &context, &mut connection).await,
+        Command::Create(create) => create::main(create, &context, &mut connection).await,
+        Command::Drop(drop) => drop::main(drop, &context, &mut connection).await,
+        Command::Wipe(wipe) => wipe::main(wipe, &context, &mut connection).await,
+        Command::List(list) => list::main(list, &context, &mut connection).await,
+        Command::Rename(rename) => rename::main(rename, &context, &mut connection).await,
+        _ => anyhow::bail!("Unknown subcommand")
```

Comment:
This `_` wildcard match is not really needed. Well, currently it is because you also have `Command::Rebase` and `Command::Merge` unimplemented, but I suggest you remove (or comment out) those until they are implemented.

Reasoning here is that "this wildcard match defeats the purpose of enums: to have a finite set of valid states and have the compiler check that you handle all of those states correctly". This is useful when you later add a new `Command` variant, and the compiler will point you to all of the locations where you are not handling that variant yet.